anitya/tests/lib/backends/test_githubapi.py

```diff
@@ -0,0 +1,157 @@
+# version.  This program is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY expressed or implied, including the
+# implied warranties of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+# PURPOSE.  See the GNU General Public License for more details.  You
+# should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# Any Red Hat trademarks that are incorporated in the source
+# code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission
+# of Red Hat, Inc.
+#
+
+'''
+anitya tests for the custom backend.
+'''
+
+import unittest
+
+import anitya.lib.backends.githubapi as backend
+import anitya.lib.model as model
+from anitya.lib.exceptions import AnityaPluginException
+from anitya.tests.base import Modeltests, create_distro, skip_jenkins
+
+
+BACKEND = 'GitHub API'
+
+
+class GithubApiBackendtests(Modeltests):
+    """ Github API backend tests. """
+
+    @skip_jenkins
+    def setUp(self):
+        """ Set up the environnment, ran before every tests. """
+        super(GithubApiBackendtests, self).setUp()
+
+        create_distro(self.session)
+        self.create_project()
+
+    def create_project(self):
+        """ Create some basic projects to work with. """
+        project = model.Project(
+            name='fedocal',
+            homepage='https://github.com/fedora-infra/fedocal',
+            version_url='fedora-infra/fedocal',
+            backend=BACKEND,
+        )
+        self.session.add(project)
+        self.session.commit()
+
+        project = model.Project(
+            name='foobar',
+            homepage='http://github.com/foo/bar',
+            version_url='foobar/bar',
+            backend=BACKEND,
+        )
+        self.session.add(project)
+        self.session.commit()
+
+        project = model.Project(
+            name='pkgdb2',
+            homepage='https://github.com/fedora-infra/pkgdb2',
+            backend=BACKEND,
+        )
+        self.session.add(project)
+        self.session.commit()
+
+    def test_get_version(self):
+        """ Test the get_version function of the github backend. """
+        pid = 1
+        project = model.Project.get(self.session, pid)
+        exp = '0.15.1'
+        obs = backend.GithubApiBackend.get_version(project)
+        self.assertEqual(obs, exp)
+
+        pid = 2
+        project = model.Project.get(self.session, pid)
+        self.assertRaises(
+            AnityaPluginException,
+            backend.GithubApiBackend.get_version,
+            project
+        )
+
+        pid = 3
+        project = model.Project.get(self.session, pid)
+        exp = '2.6.2'
+        obs = backend.GithubApiBackend.get_version(project)
+        self.assertEqual(obs, exp)
```

Comment:
Are these tests connecting to GitHub over the Internet? I ask because I didn't notice any mocking. If they connect over the Internet, these tests will fail as the referenced project make new releases.